compat/win32/pthread.c

```diff
@@ -57,6 +57,8 @@ pthread_t pthread_self(void)
 	return t;
 }
 
+#ifdef GIT_WIN_XP_SUPPORT
```

Comment:
@dscho Done!

Since every pthread_cond_* is basically one-liner, I convert the implementation to macros so that compiler can inline them.